src/gordon_gcp/plugins/enricher.py

```diff
@@ -127,19 +127,21 @@ def __init__(self, config, http_client, success_channel, error_channel):
         self._http_client = http_client
         self._logger = logging.getLogger('')
 
+    def _check_instance_data(self, instance_data):
+        assert self._get_external_ip(instance_data)
+
     async def _poll_for_instance_data(self, resource_name, msg_logger):
         exception = None
         backoff = 2
         base_url = f'https://compute.googleapis.com/compute/v1/{resource_name}'
 
-        # Poll until instance data contains external IP information.
+        # Poll until instance data contains all necessary information.
         for attempt in range(1, self.config['retries'] + 1):
             try:
                 msg_logger.debug(f'Attempt {attempt}: fetching {base_url}')
```

Comment:
@ocsw I didn't see this message was already being logged when we were going through the change together. I think this makes your added message redundant.